### ISSUE TITLE
Wrong call context binding for arrow function in interface field initializers

### ISSUE THREAD

**Issue by dbo** [NumberFour/n4js#347]
##### test.n4js.xt
```
interface I {
    num = 5;
    fn = () => console.log(this.num);
}

class C implements I {}

(new C).fn();
// XPECT output --> 5

```

I'd suggest to bind the target call context (`this`) right away when calling `$fieldInit`, if possible.

**Comment by JensN4** [comment]
It's not really about arrow functions but about wrong this binding in interface field initializers in general.

**Comment by JensN4** [comment]
Additional comment: As described in #348, this literals are not allowed in static initializers at all, and they are not allowed in general in field initializers of interfaces. We may re-define the latter decision (cf. Req. 173), but in the meantime the fix in #348 breaks the code in the example here anyway.

The original bug fix would make it easy to allow this literals in instance field initializers though. If needed, please create another bug report (feature request) to enable that. I don't remember exactly why we did not allow that in the first place, so this might need some design rethinking. I might be the case that when an interface is implemented, the order of the initializers cannot be guaranteed. That would mean using "this" in an interface field initializers may produce different results depending on how the interface is used later on. This would lead to very hard to find problems. So I would assume that it is probably better to not allow that.

**Comment by dbo** [comment]
thanks for the promptly fix!
